### ISSUE TITLE
Support .NET Standard / .NET Core

### ISSUE THREAD

**Issue by iSynaptic** [disruptor-net/Disruptor-net#26]
It appears this library only supports the full .NET Framework.  Would it be possible to switch this to support a version of the .NET Standard so this library could be used in .NET Core applications?

**Comment by Aethon** [comment]
+1

**Comment by ablanchet** [comment]
Actually it was not a priority, and I don't speak for other team members but personally I was kind of waiting for the tooling to improve on this. 

Since VS17 is now released we should be able to migrate to dotnet standard with a fairly high version number (in order to ensure the maximum compatibility with existing code). 

A PR on this would be appreciated here ;)

**Comment by ocoanet** [comment]
I tried to target .NET Standard, but some key types are missing from the current version, for example [Thread](https://apisof.net/catalog/System.Threading.Thread).

I decided to multi-target netcoreapp1.1 and net45.

[Thread.Yield](https://apisof.net/catalog/System.Threading.Thread.Yield()) is missing from .NET Core 1.1, so `YieldingWaitStrategy` will not be available, but everything else should work fine.

The .NET Core library will be included in the next nuget release.
